### PR TITLE
GOES and figure test fixes/shake-up

### PR DESCRIPTION
**Pull request by nabobalis**
Re-enabled the tests in `sunpy/net/dataretriever/tests/test_goes_ud.py`, called `test_get` and `test_new_logic`.

The fix (and I use that term lightly), was to move them to the front of the test list. This I have no idea why. **NONE**.

Hopefully (hard to test for sure) should no longer fail, in `sunpy/net/dataretriever/sources/goes.py`, called `test_time_for_url`.

The fix was to add another assume with the start of the failed example which was the date plus the time delta. 

These ran and passed locally and my local fork on Travis.

### CODE REVIEW

**Comment by Cadair** [comment]
Oh gods ordering of tests should not matter lol.

If it works it works lol ¯\_ (ツ) _/¯

**Comment by nabobalis** [comment]
Well it was strange, since If i called those tests directly from the command line via pytest, they passed. So I thought, why not order them, and they didn't hang.

It could be some cache issue or maybe the LCClient needs to be reset?

**Comment by nabobalis** [comment]
CI fail only due to figure tests.

File: sunpy/tests/figure_tests_env_2.7.yml [hunk 1]
```diff
@@ -1,77 +1,119 @@
 name: sunpy-figure-tests-2.7
 channels:
-- defaults
-- conda-forge
+  - conda-forge
+  - defaults
```
Comment (nabobalis) [inline sunpy/tests/figure_tests_env_2.7.yml]: This is the way I have it locally, should it be reversed?

**Comment by nabobalis** [comment]
I have all the figures saved locally for both runs if anyone wants me to share them.

**Comment by dstansby** [comment]
Test ordering problems might be because of the order stuff is done to `LCClient` - maybe making a new one inside each test instead of a global one in the file would help?

**Comment by Cadair** [comment]
Yeah that could be it, it also shouldn't be! Be interesting to see if making it a fixture solves it.

**Comment by nabobalis** [comment]
If by that you mean something like this?

```python
@pytest.fixture
def LCClient():
    return goes.XRSClient()
```

Not sure that's how you use the fixture method but moving the tests back and calling a new LCClinet each time seems to fix it.

**Comment by nabobalis** [comment]
NOTHING LIKE A PR GONE OUT OF CONTROL

**Comment by Cadair** [comment]
Looks good to me.

**Pull request merged**